api/v1alpha1/authorization_types.go

```diff
@@ -47,17 +50,55 @@ type Principal struct {
 	// ClientCIDRs are the IP CIDR ranges of the client.
 	// Valid examples are "192.168.1.0/24" or "2001:db8::/64"
 	//
+	// If multiple CIDR ranges are specified, one of the CIDR ranges must match
+	// the client IP for the rule to match.
+	//
 	// The client IP is inferred from the X-Forwarded-For header, a custom header,
 	// or the proxy protocol.
 	// You can use the `ClientIPDetection` or the `EnableProxyProtocol` field in
 	// the `ClientTrafficPolicy` to configure how the client IP is detected.
 	// +kubebuilder:validation:MinItems=1
 	ClientCIDRs []CIDR `json:"clientCIDRs"`
 
-	// TODO: Zhaohuabing the MinItems=1 validation can be relaxed to allow empty list
-	// after other principal types are supported. However, at least one principal is required
+	// JWTClaims are the claims in a JWT token.
+	//
+	// If multiple claims are specified, all claims must match for the rule to match.
```

Comment:
just thinking use case for two different users from two different issuers:

sub: user1 iss: issuer1
sub: user2 iss: issuer2

is there any way to allow these users? I do not think so. Basically jwtClaims should be "OR"d together and now I think it uses and everywhere?

Reply:
I mean

```
      rules:
      - action: Allow
        principal:
          jwtClaims:
          - type: string
             name: sub
             values: [user1]
          - type: string
            name: iss
            values:  [issuer1]
      - action: Allow
        principal:
          jwtClaims:
          - type: string
             name: sub
             values: [user2]
          - type: string
            name: iss
            values:  [issuer2]
```

but I think this does not work? rules should be "OR"d in this case

Reply:
ah actually, I am little bit slow

```
      rules:
      - action: Allow
        principal:
          jwtClaims:
          - type: string
             name: sub
             values: [user1, user2]
          - type: string
            name: iss
            values:  [issuer1, issuer2]
```

it might work like this. However, this now opens possible security issue. It allow also user1 from issuer2 AND user2 from issuer1

Reply:
@zetaab I think this will work because rules are evaluated in order and the first matching rule will be applied.

```
      rules:
      - action: Allow
        principal:
          jwtClaims:
          - type: string
             name: sub
             values: [user1]
          - type: string
            name: iss
            values:  [issuer1]
      - action: Allow
        principal:
          jwtClaims:
          - type: string
             name: sub
             values: [user2]
          - type: string
            name: iss
            values:  [issuer2]
```

Reply:
In that case the spec looks good